src/hub/hubUri.ts

```diff
@@ -12,17 +15,19 @@ export default async function hubUri(
     return null;
   }
 
-  let uri = await env.asExternalUri(Uri.parse(config.message));
+  const hubUrl = new URL(config.message);
 
-  // This is a prepartion for hub extraction
-  // const tabnineUrl = hubUrl.searchParams.get(TABNINE_URL_QUERY_PARAM);
-  // if (tabnineUrl)
-  //   hubUrl.searchParams.set(
-  //     TABNINE_URL_QUERY_PARAM,
-  //     (await env.asExternalUri(Uri.parse(tabnineUrl))).toString()
-  //   );
-  // let uri = Uri.parse(hubUrl.toString());
-  //
-  if (path) uri = Uri.joinPath(uri, path);
-  return uri;
+  const tabnineUrl = hubUrl.searchParams.get(TABNINE_URL_QUERY_PARAM);
+  if (tabnineUrl) {
+    hubUrl.searchParams.set(
+      TABNINE_URL_QUERY_PARAM,
+      (await env.asExternalUri(Uri.parse(tabnineUrl))).toString()
+    );
+  }
+
+  if (path) {
+    hubUrl.pathname += path.startsWith("/") ? path : `/${path}`;
```

Comment:
why not using `joinPath`?

Reply:
It since I preferred using the URL object and not the URI which doesn't support query params that well.